### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by Ghiftee**
## SEE LINK TO VIDEO PRESENTATION BELOW

> ### Project Implementations
- Splash screen

  A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign up and log in pages
  
  The user should be able to register in the app with full name, email and password (all mandatory).
  The user can log into the app using email and password.
  If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
- Home page (catagories page)
  
  When the user logs in, they are presented with the categories page.
  For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- Transactions page

  For a given category, the list of transactions is presented, ordered by the most recent.
  At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new category" page

  The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new transaction" page

  The user fills out a form to create a new transaction with:
  name (mandatory)
  amount (mandatory)
  categories (mandatory at least one)
  The user clicks (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
  When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- Testing requirements
  Create unit and integration tests for all the most important components of your RoR application.

[VIDEO PRESENTATION](https://www.loom.com/share/6e1e19ff2ee54e9186f469e48881ba27)

Kindly take a look at my project
Thank you for your time!

### CODE REVIEW

File: app/views/categories/_category.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<div class='default-font'>
+  <%= link_to category_url(category) do %>
+    <li >
+      <div class='m-3 p-4 card is-flex is-justify-content-space-between'>
+        <div class='image'>
+          <%= category.icon %>
+        </div>
```
Comment (henatan99) [inline app/views/categories/_category.html.erb:7]: - [x] In the category view element you are just rendering the icon value as a raw string, and when a user enters a huge size string, it messes up the UI. In my opinion, using the raw string as an icon is not a good idea, but if you chose that it should be like that you should at least validate the minimum string size that should be input in the form. I also suggest that you have an `img` tag for the icon which has the `src` property value equal to an image address URL, a user can input. 

image

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @Ghiftee , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ghiftee** [comment]
Thank you for the review
I have added a maximum length to the icon input so that if a user chooses an icon that's too long, it would give an error

**Review by Dipeshtwis — APPROVED** [review]
# Status: Approved ✔️

## Hi @Ghiftee,

Good job on implementing the suggested changes. Well Done. Also, Update the `live link` as well to show the latest changes.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
_**Dipesh Kumar**_

--------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**